pkg/provider/provider.go

```diff
@@ -49,6 +54,27 @@ type Legacy struct {
 func NewProviderCreator(jwtSecret string, legacy Legacy) ProviderCreator {
 	return func(user, password string) (Provider, error) {
 		switch {
+		case viper.IsSet("authorization-lambda"):
+			ok := CheckAuthorization(user, password, viper.GetString("authorization-lambda"), viper.GetString("aws-s3-region"))
+			if ok {
+				if legacy.JWT != nil {
+					origJSON, err := json.Marshal(&legacy.JWT)
+					if err != nil {
+						return nil, err
+					}
+
+					userJWT := JWT{}
+					if err = json.Unmarshal(origJSON, &userJWT); err != nil {
+						return nil, err
+					}
+					prefix := userJWT.Prefix
+					if prefix != "" && !strings.HasSuffix(prefix, "/") {
+						userJWT.Prefix += "/"
+					}
+					userJWT.Prefix += user
```

Comment:
this feels like making/mixing authentication and authorization decisions at multiple places. This behavior is not transparent to me. The lambda should make the decision which prefix the user may access. I will propose another way of delegating authentication and authorization in my final comment.

Reply:
Yes, I agree that this is less than perfect. My problem is that I'm working against an existing lambda that simply returns "OK" or an error. It is used by several services already, and it isn't clear that having it return extra information is at all meaningful for other callers.

To me, it seems that the restriction of what is visible belongs more in this domain than in the lambda.

What about an explicit "SetPrefix" method in s3.go? I can call it here after calling `newAWSS3Provider()`.

Reply:
I've thought of another possibility, which fulfills my needs but still isn't fully generalized.

What works for me is to use the `prefix` to filter messages that are visible. What if you allow the "aws-s3-prefix" to have a substitution string? Something like "Inbox/mydomain.org/{{name}}"?

This would put the control of the prefix back into the hands of the person configuring the server, not hard-coded to my specific needs. For instance, what works for someone else could be "{{name}}/Inbox".

I don't know if there are other parameters that could be of interest in this case. But please, let me know what you think of this idea. And if you are OK with it, what do you think the formatting of the substitution string should be?

Reply:
I totally get your reasoning behind this and why it might be useful for your setup. Still, the only acceptable solution for me would be a lambda that returns a JSON response that can be unmarshalled to

https://github.com/markushinz/aws-ses-pop3-server/blob/8bae93f7e7a71d4d51814141e92f2264e46506ef/pkg/provider/provider.go#L35-L41

Maybe you could come up with another lambda function that calls your existing one and provides the output as described above?